src/Database/Commands/DeleteByIndexCommand.ts

```diff
@@ -0,0 +1,36 @@
+import {ServerNode} from '../../Http/ServerNode';
+import {RequestMethods} from "../../Http/Request/RequestMethod";
+import {ErrorResponseException} from "../DatabaseExceptions";
+import {StringUtil} from "../../Utility/StringUtil";
+import {IndexQuery} from "../Indexes/IndexQuery";
+import {QueryOperationOptions} from "../Operations/QueryOperationOptions";
+import {IResponse, IResponseBody} from "../../Http/Response/IResponse";
+import {IRavenResponse} from "../RavenCommandResponse";
+import {StatusCode, StatusCodes} from "../../Http/Response/StatusCode";
+import {IndexQueryBasedCommand} from "./IndexQueryBasedCommand";
+
+export class DeleteByIndexCommand extends IndexQueryBasedCommand {
+  constructor(indexName: string, query: IndexQuery, options?: QueryOperationOptions) {
+    super(RequestMethods.Delete, indexName, query, options);
+  }
+
+  public createRequest(serverNode: ServerNode): void {
+    this.endPoint = StringUtil.format('{url}/databases/{database}', serverNode, this.params);
+    super.createRequest(serverNode);
+  }
+
+  public setResponse(response: IResponse): IRavenResponse | IRavenResponse[] | null | void {
+    const responseBody: IResponseBody = response.body;
+    const status: StatusCode = response.statusCode;
+
```

Comment:
status code is HTTP status code (from http request to raven db server)
it doesn't includes to responses in python client, it's used only to determine kind of error and for to throw corresponding exception